totp.go

```diff
@@ -1,62 +1,84 @@
 package walletclient
 
 import (
+	"encoding/base32"
+	"encoding/hex"
 	"errors"
 	"fmt"
+	"time"
 
-	"github.com/bitcoin-sv/spv-wallet-go-client/totp"
 	"github.com/bitcoin-sv/spv-wallet-go-client/utils"
 	"github.com/bitcoin-sv/spv-wallet/models"
 	"github.com/bitcoinschema/go-bitcoin/v2"
+	"github.com/libsv/go-bk/bec"
 	"github.com/libsv/go-bk/bip32"
+	"github.com/pquerna/otp"
+	"github.com/pquerna/otp/totp"
 )
 
-// GenerateTotpForContact creates one time-based one-time password based on secret shared between the user and the contact
-func (b *WalletClient) GenerateTotpForContact(contact *models.Contact, validationPeriod, digits uint) (string, error) {
-	if b.xPriv == nil {
-		return "", errors.New("init client with xPriv first")
-	}
+var ErrClientInitNoXpriv = errors.New("init client with xPriv first")
 
-	xpriv, err := deriveXprivForPki(b.xPriv)
+// GenerateTotpForContact creates one time-based one-time password based on secret shared between the user and the contact
+func (b *WalletClient) GenerateTotpForContact(contact *models.Contact, period, digits uint) (string, error) {
```

Comment:
At least we should have some defaults for the period and digits.
Or if there already is something in the library underneath then we should document those defaults on this method.